presto-main/src/main/java/io/prestosql/sql/planner/iterative/rule/RemoveUnsupportedDynamicFilters.java

```diff
@@ -181,12 +187,41 @@ private Expression removeDynamicFilters(Expression expression, Set<String> allow
 
         private Expression removeAllDynamicFilters(Expression expression)
         {
-            DynamicFilters.ExtractResult extractResult = extractDynamicFilters(expression);
+            Expression rewrittenExpression = removeNestedDynamicFilters(expression);
+            DynamicFilters.ExtractResult extractResult = extractDynamicFilters(rewrittenExpression);
             if (extractResult.getDynamicConjuncts().isEmpty()) {
-                return expression;
+                return rewrittenExpression;
             }
             return combineConjuncts(metadata, extractResult.getStaticConjuncts());
         }
+
+        private Expression removeNestedDynamicFilters(Expression expression)
+        {
+            return ExpressionTreeRewriter.rewriteWith(new RemoveDynamicFilters(), expression);
+        }
+    }
+
+    private static class RemoveDynamicFilters
+            extends io.prestosql.sql.tree.ExpressionRewriter<Void>
+    {
+        @Override
+        public Expression rewriteLogicalBinaryExpression(LogicalBinaryExpression node, Void context, ExpressionTreeRewriter<Void> treeRewriter)
+        {
+            LogicalBinaryExpression rewrittenNode = treeRewriter.defaultRewrite(node, context);
+
+            ImmutableList.Builder<Expression> expressionBuilder = ImmutableList.builder();
+            if (!isDynamicFilter(rewrittenNode.getLeft())) {
+                expressionBuilder.add(rewrittenNode.getLeft());
+            }
+            if (!isDynamicFilter(rewrittenNode.getRight())) {
+                expressionBuilder.add(rewrittenNode.getRight());
+            }
+            List<Expression> staticPredicates = expressionBuilder.build();
+            if ((node == rewrittenNode) && staticPredicates.size() == 2) {
+                return node;
+            }
+            return binaryExpression(node.getOperator(), staticPredicates);
```

Comment:
this is probably wrong. For example:
```
DF(x) OR x IS NULL 
```
should be rewritten to:
```
true OR X is NULL equivalent of TRUE
```
Similarly:
DF(x) AND x IS NULL
should be rewritten to:
```
TRUE AND x IS NULL equivalent of x IS NULL
```

Please add a test case

Reply:
Fixed and added `TestRemoveUnsupportedDynamicFilters`